lib/semantics/symbol.cc

```diff
@@ -247,6 +247,15 @@ bool Symbol::CanReplaceDetails(const Details &details) const {
   }
 }
 
+// Usually a symbol's name is the first occurrence in the source, but sometimes
+// we want to replace it with one at a different location (but same characters).
+void Symbol::ReplaceName(const SourceName &name) {
+  CHECK(name == name_);
+  if (name.begin() != name_.begin()) {
```

Comment:
You can just assign `name_ = name;` unconditionally.

Reply:
Yes, good point.